service/history/workflow/mutable_state_state_status.go

```diff
@@ -135,7 +135,7 @@ func setStateStatus(
 			}
 
 		case enumsspb.WORKFLOW_EXECUTION_STATE_ZOMBIE:
-			if status == enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
+			if status != enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING {
```

Comment:
why? Zombie is not running correct? https://github.com/temporalio/temporal/blob/6691620a207b6b9245879adad0ffb1478aac2f3b/service/history/workflow/mutable_state_impl.go#L1533

Reply:
this is a bug fix, although this path is not utilized